docs/GetStarted/getting_started_macos_cmake.md

```diff
@@ -0,0 +1,120 @@
+# Getting Started on MacOS with CMake
+
+<!--
+Notes to those updating this guide:
+
+    * This document should be __simple__ and cover essential items only.
+      Notes for optional components should go in separate files.
+
+    * This document parallels getting_started_linux_cmake.md and
+      getting_started_windows_cmake.md
+      Please keep them in sync.
+-->
+
+This guide walks through building the core compiler and runtime parts of IREE
+from source. Auxiliary components like the Python bindings and Vulkan driver are
+not documented for MacOS at this time.
+
+IREE is not officially supported on MacOS at this time. It may work, but it is
+not a part of our open source CI, and may be intermittently broken.
+Contributions related to MacOS support and documentation are welcome however.
+
+## Prerequisites
+
+### Install Homebrew
+
+This guide uses [Homebrew](https://brew.sh/) to install IREE's dependencies.
+
+```shell
+$ /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
+```
+
+### Install CMake
+
+IREE uses [CMake](https://cmake.org/) version `>= 3.13`. Brew ships the latest
+release.
+
+```shell
+$ brew install cmake
+$ cmake --version  # >= 3.13
+```
+
+### Install Ninja
+
+[Ninja](https://ninja-build.org/) is a fast build system that you can use as a
+CMake generator.
+
+```
+$ brew install ninja
+```
+
+### Install a Compiler
+
+We recommend Clang. GCC is not fully supported. Appropriate versions of `clang`
+and `clang++` ship with MacOS by default.
+
+## Clone and Build
+
+### Clone
+
+Clone the repository and initialize its submodules:
+
+```shell
+$ git clone https://github.com/google/iree.git
+$ cd iree
+$ git submodule update --init
+```
+
+> Tip:<br>
+> &nbsp;&nbsp;&nbsp;&nbsp;Editors and other programs can also clone the
+> repository, just make sure that they initialize the submodules.
+
+### Build
+
+Configure:
+
+```shell
+$ cmake -G Ninja -B build/ -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ .
```

Comment:
I think we don't need the explicit compiler specification?

Reply:
Looks like we don't anymore. I updated the linux doc as well.

Reply:
If you don't pass `-DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++` on Linux, by default it should finds gcc on most distributions:

```
mbrehler@.. ~/repos/iree/build$ cmake -GNinja  ..
-- The CXX compiler identification is GNU 9.3.0
-- The C compiler identification is GNU 9.3.0
...
```

Thus, the explicit compiler specification shouldn't be removed from the Linux doc!

Reply:
Good catch! making the fix.